pkg/gofr/http/middleware/logger.go

```diff
@@ -112,3 +114,38 @@ func panicRecovery(w http.ResponseWriter, logger logger) {
 		_ = json.NewEncoder(w).Encode(res)
 	}
 }
+
+func GetCorrelationID(r *http.Request) string {
```

Comment:
Instead of modifying the request header directly, can we return (string,error) from GetCorrelationID and set the headers where it is being called.. ?
Or think of moving it to some other location where it can access the request easily.